File: pe-core/src/main/java/org/saar/core/model/ArraysMesh.java
package org.saar.core.model;

import org.saar.lwjgl.opengl.constants.RenderMode;
import org.saar.lwjgl.opengl.objects.vaos.IVao;
import org.saar.lwjgl.opengl.utils.GlRendering;

public class ArraysMesh extends MeshBase implements Mesh {

    private final RenderMode mode;
    private final int count;

    public ArraysMesh(IVao vao, RenderMode mode, int count) {
        super(vao);
        this.mode = mode;
        this.count = count;
    }

    @Override
    protected void doDrawCall() {
        GlRendering.drawArrays(this.mode, 0, this.count);
    }
}

File: pe-core/src/main/java/org/saar/core/mesh/ArraysMesh.kt
package org.saar.core.mesh

import org.saar.lwjgl.opengl.constants.RenderMode
import org.saar.lwjgl.opengl.drawcall.ArraysDrawCall
import org.saar.lwjgl.opengl.vao.IVao

class ArraysMesh(
    private val vao: IVao,
    var drawCall: ArraysDrawCall,
) : Mesh {

    fun set(
        renderMode: RenderMode = this.drawCall.renderMode,
        first: Int = this.drawCall.first,
        count: Int = this.drawCall.count,
    ) {
        this.drawCall = ArraysDrawCall(renderMode, first, count)
    }

    override fun draw() {
        this.vao.bind()
        this.drawCall.doDrawCall()
    }

    override fun delete() {
        this.vao.delete()
    }
}